packages/sdk/src/BaseAccountAPI.ts

```diff
@@ -280,6 +310,66 @@ export abstract class BaseAccountAPI {
     }
   }
 
+  /**
+   * create a UserOperation for a batch transaction, filling all details (except signature)
+   * - if account is not yet created, add initCode to deploy it.
+   * - if gas or nonce are missing, read them from the chain (note that we can't fill gaslimit before the account is created)
+   * @param info
+   */
+  async createUnsignedUserOpBatch (info: BatchTransactionDetailsForUserOp): Promise<UserOperationStruct> {
```

Comment:
this method shares 90% of the code of createUnsignedUserOp.
the only difference is the "encode".
Seems better to remove it, and make "target" and "data" params of the original method to be `string | string[]` - that is, accept either an array or single value.
the encoding logic is moved into encodeUserOpCallDataAndGasLimit

when encoding, should check their actual type, and do the right encoding.
e.g.
```
if (typeof data === 'string') {
  if (typeof target !=  'string') throw new Error ..
  use encodeExecute
} else {
  do encodeBatch
(support single "target" and multiple "data", as it seems a useful case...
}

```